gallery/how_to/work_with_microtvm/micro_ethosu.py

```diff
@@ -0,0 +1,1014 @@
+#     #define ETHOSU_NPU_FAIL (1 << 4)
+#     #define ETHOSU_IRQ ((IRQn_Type)56)
+#     #define ETHOSU_BASE_ADDRESS ((void*)0x48102000)
+#
+#     #endif  // TVM_APPS_MICROTVM_ETHOS_U_ETHOSU_55_H_
+#
+# .. code-block:: c
+#    :caption: ethosu_mod.h
+#    :name: ethosu_mod.h
+#
+#     #ifndef TVM_APPS_MICROTVM_ETHOS_U_ETHOSU_MOD_H_
+#     #define TVM_APPS_MICROTVM_ETHOS_U_ETHOSU_MOD_H_
+#
+#     #include <ARMCM55.h>
+#     #include <ethosu_driver.h>
+#     #include <stdio.h>
+#
+#     #include "ethosu_55.h"
+#
+#     struct ethosu_driver* ethosu0_driver = &ethosu_drv;
+#
+#     void ethosuIrqHandler0() { ethosu_irq_handler(ethosu0_driver); }
+#
+#     // Initialize Ethos(TM)-U NPU driver
+#     int EthosuInit() {
+#       if (ethosu_init(ethosu0_driver, (void*)ETHOSU_BASE_ADDRESS, NULL, 0, 1, 1)) {
+#         printf("Failed to initialize NPU.\n");
+#         return -1;
+#       }
+#
+#       // Assumes SCB->VTOR points to RW memory
+#       NVIC_SetVector(ETHOSU_IRQ, (uint32_t)&ethosuIrqHandler0);
+#       NVIC_EnableIRQ(ETHOSU_IRQ);
+#
+#       return 0;
+#     }
+#
+#     #endif  // TVM_APPS_MICROTVM_ETHOS_U_ETHOSU_MOD_H_
+#
+#
+# .. code-block:: c
+#    :caption: crt_config.h
+#    :name: crt_config.h
+#
+#     #ifndef TVM_RUNTIME_CRT_CONFIG_H_
+#     #define TVM_RUNTIME_CRT_CONFIG_H_
+#
+#     /*! Log level of the CRT runtime */
+#     #define TVM_CRT_LOG_LEVEL TVM_CRT_LOG_LEVEL_DEBUG
+#
+#     #endif  // TVM_RUNTIME_CRT_CONFIG_H_
+#
+#
+# .. code-block:: c
+#    :caption: tvm_runtime.h.h
+#    :name: tvm_runtime.h.h
+#
+#     #include <stdarg.h>
+#     #include <stdio.h>
+#     #include <stdlib.h>
+#     #include <tvm/runtime/c_runtime_api.h>
+#     #include <tvm/runtime/crt/stack_allocator.h>
+#
+#     #ifdef __cplusplus
+#     extern "C" {
+#     #endif
+#
+#     #define WORKSPACE_SIZE (16384 * 1024)
+#     __attribute__((section("ethosu_scratch"))) static uint8_t g_aot_memory[WORKSPACE_SIZE];
+#
+#     tvm_workspace_t app_workspace;
+#
+#     void __attribute__((noreturn)) TVMPlatformAbort(tvm_crt_error_t error_code) {
+#       printf("TVMPlatformAbort: %d\n", error_code);
+#       printf("EXITTHESIM\n");
+#       exit(-1);
+#     }
+#
+#     tvm_crt_error_t TVMPlatformMemoryAllocate(size_t num_bytes, DLDevice dev, void** out_ptr) {
+#       return StackMemoryManager_Allocate(&app_workspace, num_bytes, out_ptr);
+#     }
+#
+#     tvm_crt_error_t TVMPlatformMemoryFree(void* ptr, DLDevice dev) {
+#       return StackMemoryManager_Free(&app_workspace, ptr);
+#     }
+#
+#     void TVMLogf(const char* msg, ...) {
+#       va_list args;
+#       va_start(args, msg);
+#       vfprintf(stdout, msg, args);
+#       va_end(args);
+#     }
+#
+#     TVM_DLL int TVMFuncRegisterGlobal(const char* name, TVMFunctionHandle f, int override) { return 0; }
+#
+#     #ifdef __cplusplus
+#     }
+#     #endif
+
+################################################################################
+# Creating the linker script
```

Comment:
just wondering if we should include stuff like the linker script here or just link to it in the codebase from tutorial? it's a bit low level is all. on the other hand it is quite informative to understand the system.